src/rule.rs

```diff
@@ -37,6 +37,16 @@ impl RuleBuilder {
     }
 }
 
+#[macro_export]
+macro_rules! rule {
+    ($($variant:ident => $value:expr),* $(,)?) => {
+        proc_macros::builder_args!(
```

Comment:
Since at the moment these are both just direct mappings to the underlying builder methods, delegate the work to a generic `builder_args!()` proc macro